### PR TITLE
VZ-3026: Optional CA Secret

### PR DESCRIPTION
**Pull request by stevshap**
# Description

This PR makes the CASecret field optional for a managed cluster when using a well-known CA (reflected in `vmc_types.go`). This change caused the multicluster webhook to be updated along with its corresponding test suite. When a caSecret is omitted, a warning is logged.

Unit tests in `vmc_controller_tests` were also updated to pick up the change, with the addition of a new caSecret boolean in functions `expectVmcGetAndUpdate` and `expectSyncPrometheusScraper`. Code coverage was increased to verify the creation of a job in the admin cluster's Prometheus scrape config map associated with its' managed cluster.

Fixes VZ-3026

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/apis/clusters/v1alpha1/verrazzanomanagedcluster_webhook.go [hunk 1]
```diff
@@ -109,8 +108,12 @@ func newScheme() *runtime.Scheme {
 
 // validateSecret enforces that the CA secret name was specified and that the secret exists
 func (v VerrazzanoManagedCluster) validateSecret(client client.Client) error {
+	log := zap.S().With("source", "webhook", "operation", "update", "resource", fmt.Sprintf("%s:%s", v.Namespace, v.Name))
+	log.Info("Validate secret")
```
Comment (bobdonat) [inline platform-operator/apis/clusters/v1alpha1/verrazzanomanagedcluster_webhook.go:112]: is this debug info that is not needed

File: platform-operator/controllers/clusters/sync_prometheus.go [hunk 2]
```diff
@@ -48,18 +48,19 @@ basic_auth:
 // syncPrometheusScraper will create a scrape configuration for the cluster and update the prometheus config map.  There will also be an
 // entry for the cluster's CA cert added to the prometheus config map to allow for lookup of the CA cert by the scraper's HTTP client.
 func (r *VerrazzanoManagedClusterReconciler) syncPrometheusScraper(ctx context.Context, vmc *clustersv1alpha1.VerrazzanoManagedCluster) error {
-	// read the configuration secret specified
-	if len(vmc.Spec.CASecret) == 0 {
-		return nil
-	}
-
 	var secret corev1.Secret
 	secretNsn := types.NamespacedName{
```
Comment (bobdonat) [inline platform-operator/controllers/clusters/sync_prometheus.go:52]: can this be moved to inside the if statement below that is where it is used

File: platform-operator/apis/clusters/v1alpha1/verrazzanomanagedcluster_types.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ type VerrazzanoManagedClusterSpec struct {
 
 	// The name of a secret that contains the ca certificate for accessing console
 	// and api endpoints on the managed cluster.
-	CASecret string `json:"caSecret"`
+	CASecret string `json:"caSecret,omitempty"`
```
Comment (pfmackin) [inline platform-operator/apis/clusters/v1alpha1/verrazzanomanagedcluster_types.go:21]: you need to do the following and add the resulting generated code to the PR
make generate
make manifests

File: platform-operator/apis/clusters/v1alpha1/verrazzanomanagedcluster_webhook.go [hunk 4]
```diff
@@ -18,6 +16,7 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
+	"net/url"
```
Comment (mgianatagh) [inline platform-operator/apis/clusters/v1alpha1/verrazzanomanagedcluster_webhook.go:19]: The imports are not organized properly.  Do you have the "File Watcher" enabled and configured for "goimports" and "go fmt"?

File: platform-operator/controllers/clusters/vmc_controller_test.go [hunk 5]
```diff
@@ -249,6 +251,65 @@ func TestCreateVMCOCIDNS(t *testing.T) {
 	asserts.Equal(time.Duration(0), result.RequeueAfter)
 }
 
+// TestCreateVMC tests the Reconcile method for the following use case
```
Comment (mgianatagh) [inline platform-operator/controllers/clusters/vmc_controller_test.go:254]: TestCreateVMCNoCACert

**Comment by mgianatagh** [comment]
The documentation will need to be updated after this PR is merged.

File: platform-operator/apis/clusters/v1alpha1/verrazzanomanagedcluster_webhook.go (hunk 1 above)
Comment (stevshap) [reply platform-operator/apis/clusters/v1alpha1/verrazzanomanagedcluster_webhook.go:112]: I'm actually not sure what this is doing, I pulled it in from a similar function to initialize the log ..

File: platform-operator/apis/clusters/v1alpha1/verrazzanomanagedcluster_types.go (hunk 3 above)
Comment (stevshap) [reply platform-operator/apis/clusters/v1alpha1/verrazzanomanagedcluster_types.go:21]: got it thx

File: platform-operator/controllers/clusters/vmc_controller_test.go (hunk 5 above)
Comment (stevshap) [reply platform-operator/controllers/clusters/vmc_controller_test.go:254]: good catch

**Review by mgianatagh — APPROVED** [review]

**Pull request merged**